packages/modules/product/src/repositories/product-category.ts

```diff
@@ -20,12 +20,15 @@ export class ProductCategoryRepository extends DALUtils.MikroOrmBaseTreeReposito
     familyOptions: ProductCategoryTransformOptions = {}
   ) {
     const findOptions_ = { ...findOptions }
-    findOptions_.options ??= {
-      orderBy: { rank: "ASC" },
+    findOptions_.options.orderBy = {
```

Comment:
note: in the tests, the fields where provided and hence the options always populated, meaning that the order was never applied. By some random luck from god, the order remain the same for a long time, but the truth is that without any ordering we cant guarantee the order of the retrieved data. Now we always apply both rank and id, because rank only would also randomize all categories with the same rand and hence not enforce any ordering among each rank value. Now all categories are always ordered and the order remain constant.